api/CanMsg.h

```diff
@@ -82,11 +92,45 @@ class CanMsg : public Printable
     return n;
   }
 
+
+  uint32_t getStandardId() const {
+    return (id & CAN_SFF_MASK);
+  }
+  uint32_t getExtendedId() const {
+    return (id & CAN_EFF_MASK);
+  }
+  bool isStandardId() const {
+    return ((id & CAN_EFF_FLAG) == 0);
+  }
+  bool isExtendedId() const {
+    return ((id & CAN_EFF_FLAG) == CAN_EFF_FLAG);
+  }
+
+
+  /*
+   * CAN ID semantics (mirroring definition by linux/can.h):
+   * |- Bit   31 : frame format flag (0 = standard 11 bit, 1 = extended 29 bit)
+   * |- Bit   30 : reserved (future remote transmission request flag)
+   * |- Bit   29 : reserved (future error frame flag)
+   * |- Bit 0-28 : CAN identifier (11/29 bit)
+   */
   uint32_t id;
   uint8_t  data_length;
   uint8_t  data[MAX_DATA_LENGTH];
 };
 
+/**************************************************************************************
+ * FREE FUNCTIONS
```

Comment:
So...I have to pay for the functions above?

Reply:
Free functions refers to stand-alone functions outside a class, see [here](https://stackoverflow.com/questions/4861914/what-is-the-meaning-of-the-term-free-function-in-c) :wink: